Source/JavaScriptCore/wasm/WasmOperations.cpp

```diff
@@ -932,6 +932,11 @@ JSC_DEFINE_JIT_OPERATION(operationWasmArrayNew, EncodedJSValue, (Instance* insta
     return arrayNew(instance, typeIndex, size, encValue);
 }
 
+JSC_DEFINE_JIT_OPERATION(operationWasmArrayNewData, EncodedJSValue, (Instance* instance, uint32_t typeIndex, uint32_t dataSegmentIndex, uint32_t arraySize, uint32_t offset))
+{
```

Comment:
You need to put 

```
    CallFrame* callFrame = DECLARE_WASM_CALL_FRAME(instance);
    VM& vm = instance->vm();
    NativeCallFrameTracer tracer(vm, callFrame);
```

If the following operations can cause GC / error etc. And in this case, they can happen.